### ISSUE TITLE
[red-knot] Check __init__ arguments on class literals call

### ISSUE THREAD

**Issue by mishamsk** [astral-sh/ruff#16511]
## Description

Currently red knot assumes all calls to class literals are valid and create instances. E.g. [this test](https://github.com/astral-sh/ruff/blob/bb44926ca5555b3dce59d1c14d5dbecee3cf9825/crates/red_knot_python_semantic/resources/mdtest/unary/invert_add_usub.md?plain=1#L8C4-L21C13) yields no diagnostic currently

```py
from typing import Literal

class Number:
    def __init__(self, value: int):
        self.value = 1

    def __pos__(self) -> int:
        return +self.value

    def __neg__(self) -> int:
        return -self.value

    def __invert__(self) -> Literal[True]:
        return True

a = Number()
```

We should check `__init__` signature just like any other call and emit diagnostics.

This is step 1 in a series of feature to support the instance creation process in full. The related logic of handling `type[C]` is tracked in #15948
